src/stepfunctions/steps/sagemaker.py

```diff
@@ -159,6 +159,8 @@ def get_expected_model(self, model_name=None):
             model.name = model_name
         else:
             model.name = self.job_name
+        if self.estimator.environment:
+            model.env = self.estimator.environment
```

Comment:
CR description should also capture why we're doing this. It's not strictly related to the bug reported, but is solving an issue. Typically better to fix things separately, but when there are multiple fixes, they need to be captured in the commit summary, along with rationale and any testing that was performed.

Reply:
Updated the description with the uncovered bug and what was done to fix it
I am open to move this fix to a separate PR - will go with what reviewers prefer since the review process has already begun

What do you think? @shivlaks  @wong-a

Reply:
Given where we currently are, i'm not strongly opinionated because it feels more practical to go with it. It feels like it might just be busy work splitting them up in the current state.

Generally and going forward, I think we should keep changes contained and specific to the bug they address / feature they introduce for a few reasons:
* reverts are simpler if there's a problem with one fix and not with another
* the discovered bug may not have been triaged with a process of repro steps, expected behaviour, actual behaviour to converge on a candidate fix.
* simpler for reviewers and contributors - simplifies life for everyone involved
* helps to clearly see what was fixed for inclusion in changelog when we perform next release.
* this PR grows the scope of the change that it supersedes

Reply:
If it's already implemented and works, let's keep it in here, since it's a small enough change. Just update the PR description accordingly.

Reply:
Done :)